src/lib.rs

```diff
@@ -383,27 +387,27 @@ fn run_command(
 
                 match options.restart {
                     RestartOptions::Continue => {
-                        return Some(status);
+                        return (command_name, Some(status));
```

Comment:
You have `command` that contains the CLI command running here (as an `InnerCommand` instance). You could join the `command` and `args` properties into the full command string. Maybe add a `to_arg` method to `InnerCommand`?

Reply:
I can add that in but it might not be the way it was entered into the constructor. 
For example, if a user called `Command::from_string("name", "sh -c \"echo hi\""). So then I'd have to try and detect if quotes are necessary or not. 

Or if a user split up the args themself like: `["sh", "-c", "echo hi"]`. Do I add quotes to the string here?

Reply:
Aren't these rules very simple? Like if a value contains a space, add quotes around it? 

But I agree it might be too much fumbling with data here. It's not clean. 

Another option might to carry the original command (given by the caller) around in the `InnerCommand` struct. Not because it's technically needed or elegant but because it helps create better error messages. Good error messages create a good amount of extra complexity.

Not sure if you want to make these tradeoffs here. I think it's okay if you just return the name of the failed command and let the client do the lookup. Just thinking loud here.

Reply:
Passing the original command around doesn't help if the user uses the `argv` interface to create a command. So I'm tempted to leave it as is (just the command name) for now

Reply:
Yes, makes sense. :+1: